src/main/java/at/hannibal2/skyhanni/test/command/TrackParticlesCommand.kt

```diff
@@ -71,6 +88,9 @@ object TrackParticlesCommand {
         }
     }
 
+    fun EnumParticleTypes.getByIdOrNull(name: String): EnumParticleTypes? =
+        EnumParticleTypes.entries.firstOrNull { it.name.equals(name, ignoreCase = false) }
```

Comment:
```suggestion
    private fun getParticleTypeByName(name: String): EnumParticleTypes? =
        EnumParticleTypes.entries.firstOrNull { it.name.equals(name, ignoreCase = true) }
```

I feel this is more appropriate than an extension function that has to be called on an entry of the enum as then above you can just do
`val type = getParticleTypeByName(name)`

This also makes it easier to get this working on 1.21

Reply:
good catch, `EnumParticleTypes.CRIT.getByIdOrNull(name)` this line above is obviously not intended to say.